File: 01 Two Sum/two_sum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target){
    unordered_map<int, int> map;

    for(int i = 0; i<nums.size(); i++){
        int complement = target - nums[i];
        if(map.find(complement) != map.end()){
            return {map[complement], i};
        }
        map[nums[i]] = i;
    }
    return {};
    }
}; 

int main(){
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
   
    Solution test;
    vector<int> ans = test.twoSum(nums, target);
    cout <<"Output : [" << ans[0] << ", " << ans[1] << "]" <<endl;

    return 0;
}

File: 01 Two Sum/two_sum.js
var twoSum = function (numbers, target) {
  let map = new Map();
  for (let i = 0; i < numbers.length; i++) {
    let remain = target - numbers[i];
    if (map.has(remain)) {
      return [map.get(remain), i];
    }
    map.set(numbers[i], i);
  }
  return [];
};

console.log("Test 1: numbers = [2,7,11,15], target = 9");
console.log("Result:", twoSum([2,7,11,15], 9));
console.log("Expected: [0, 1]");

console.log("\nTest 2: numbers = [3,2,4], target = 6");
console.log("Result:", twoSum([3,2,4], 6));
console.log("Expected: [1, 2]");

console.log("\nTest 3: numbers = [3,3], target = 6");
console.log("Result:", twoSum([3,3], 6));
console.log("Expected: [0, 1]");